universal-application-tool-0.0.1/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -163,13 +178,11 @@ private PredicateValue parsePredicateValue(Scalar scalar, Operator operator, Str
             return PredicateValue.of(Long.parseLong(value));
         }
 
-      default: // STRING, LIST_OF_STRINGS
+      default: // STRING, should not include LIST_OF_STRINGS since that should be covered at the top
+        // of the method with `values`.
         switch (operator) {
           case ANY_OF:
           case NONE_OF:
-          case IN:
```

Comment:
question: I'm not sure I understand why these operators disappear. I thought all of these have a list of strings on the right?

Reply:
They do all have list of strings on the right, that is correct! Let me see if I can explain this effectively in writing.

Once we get to this point, we know the scalar on the left is of type STRING, and that it is *not* of value SELECTION. That's because I added that `if` statement at the top that parses the `values` list rather than the `value` string. I updated the logic in that if statement so it's hopefully a little more clear what's going on.

Additionally, any `STRING` scalar type only has `ANY_OF`, `NONE_ON`, `EQUAL_TO`, `NOT_EQUAL_TO` as operator options. The other ones (`IN`, etc) are not available unless there is a `LIST_OF_STRINGS` on the left hand side.

The outer switch statement is based on the scalar on the *left hand side*. I know this is kind of collaged together and awkward, but hopefully it works. :)